pandas/tseries/frequencies.py

```diff
@@ -298,6 +307,23 @@ def to_offset(freqstr):
             name, stride = stride, name
         name, _ = _base_and_stride(name)
         delta = get_offset(name) * stride
+
+    elif isinstance(freqstr, timedelta):
+        delta = None
+        freqstr = tslib.Timedelta(freqstr)
```

Comment:
just add the ``Timedelta`` import ``from pandas`` at the top, rather than using ``tslib`` (its the same but prefer this way of importing)